kernel/src/process_standard.rs

```diff
@@ -258,6 +276,64 @@ impl<C: Chip> Process for ProcessStandard<'_, C> {
         ret
     }
 
+    /// Transition an unverified process into the State::Unstarted state.
+    /// Should require a capability. -pal
+    fn mark_credentials_pass(
+        &self,
+        credentials: Option<TbfFooterV2Credentials>,
+        short_app_id: Option<ShortID>,
+        _capability: &dyn capabilities::ProcessApprovalCapability,
+    ) -> Result<(), ErrorCode> {
+        if self.state.get() != State::Unchecked {
+            Err(ErrorCode::NODEVICE)
+        } else {
+            self.state.update(State::Unstarted);
+            self.app_id.insert(short_app_id);
+            credentials.map(|c| self.credentials.replace(c));
+            Ok(())
+        }
+    }
+
+    fn mark_credentials_fail(&self, _capability: &dyn capabilities::ProcessApprovalCapability) {
+        self.state.update(State::CredentialsFailed);
+    }
+
+    fn get_credentials(&self) -> Option<TbfFooterV2Credentials> {
+        let c = self.credentials.take();
+        self.credentials.insert(c);
+        c
+    }
+
+    /// Enqueue the initialization function of a process onto its task list;
+    /// this is used to start a process. Should only be called when a process
+    /// is in the `State::Unstarted` state.
+    fn enqueue_init_task(
+        &self,
+        _cap: &dyn capabilities::ProcessInitCapability,
+    ) -> Result<(), ErrorCode> {
+        if self.state.get() != State::Unstarted && self.state.get() != State::Terminated {
+            return Err(ErrorCode::NODEVICE);
+        }
+
+        self.state.update(State::Yielded);
+
+        // And queue up this app to be restarted.
+        let flash_start = self.flash_start();
+        let app_start =
+            unsafe { flash_start.offset(self.header.get_app_start_offset() as isize) as usize };
+        let init_fn =
+            unsafe { flash_start.offset(self.header.get_init_function_offset() as isize) as usize };
+
+        self.enqueue_task(Task::FunctionCall(FunctionCall {
+            source: FunctionCallSource::Kernel,
+            pc: init_fn,
+            argument0: app_start as usize,
+            argument1: self.memory_start as usize,
+            argument2: self.memory_len,
+            argument3: self.app_break.get() as usize,
+        }))
+    }
```

Comment:
I think this function needs a better specified return type, and needs to clarify what state the process is in if it returns an `Err()`. If this fails, I don't think the process should be `Yielded`.

Reply:
Shouldn't it be Yielded, but with work to do? Technically the process is blocked until an upcall; that upcall happens to be the init function. Or do you think there should be a special state for "waiting for init to be invoked"? If you think it would make things conceptually clearer we can add a new state, but functionally I think it matches Yielded?